ConformanceChecker: Make `actor_isolated_witness` call out the protocol

AnthonyLatsis opened the pull request:
Resolves #78518.

AnthonyLatsis committed:
ConformanceChecker: Make `actor_isolated_witness` call out the protocol

xedin commented on lib/Sema/TypeCheckProtocol.cpp:
```diff
@@ -3444,10 +3444,12 @@ ConformanceChecker::checkActorIsolation(ValueDecl *requirement,
+                 isDistributed && !isDistributedDecl(witness),
+                 refResult.isolation, witness, requirementIsolation,
+                 Conformance->getProtocol())
+      .limitBehaviorUntilSwiftVersion(behavior, 6);
```
Should we also add a note to point at the requirement?

AnthonyLatsis replied:
Ah, we already emit one:

https://github.com/swiftlang/swift/blob/b718c6d448104d7365ebf1c48eb5f0605631ab19/lib/Sema/TypeCheckProtocol.cpp#L3535

But "x declared here" is not as helpful and clear as I would like it to be, especially when there are other notes attached. I will swap it for a new note that says "requirement x declared here".

xedin replied:
Sounds good! Could be done separately as well.

AnthonyLatsis committed:
Sema: Improve requirement indicator note for `actor_isolated_witness`

`x declared here` is not helpful and clear enough, especially when there
are other notes attached. Swap it for a new note that says
`requirement x declared here`.

AnthonyLatsis commented:
@swift-ci please smoke test

AnthonyLatsis commented:
@swift-ci please smoke test
